pkg/metrics/metrics.go

```diff
@@ -1224,22 +1251,26 @@ func CreateConfiguration(metricsEnabled []string) (Configuration, []prometheus.C
 
 		case Namespace + "_" + SubsystemKVStore + "_operations_duration_seconds":
 			KVStoreOperationsDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
-				Namespace: Namespace,
-				Subsystem: SubsystemKVStore,
-				Name:      "operations_duration_seconds",
-				Help:      "Duration in seconds of kvstore operations",
+				Namespace:                      Namespace,
+				Subsystem:                      SubsystemKVStore,
+				Name:                           "operations_duration_seconds",
+				Help:                           "Duration in seconds of kvstore operations",
+				NativeHistogramBucketFactor:    HistogramFactor,
+				NativeHistogramMaxBucketNumber: HistogramMaxBuckets,
 			}, []string{LabelScope, LabelKind, LabelAction, LabelOutcome})
 
 			collectors = append(collectors, KVStoreOperationsDuration)
 			c.KVStoreOperationsDurationEnabled = true
 
 		case Namespace + "_" + SubsystemKVStore + "_events_queue_seconds":
 			KVStoreEventsQueueDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
-				Namespace: Namespace,
-				Subsystem: SubsystemKVStore,
-				Name:      "events_queue_seconds",
-				Help:      "Duration in seconds of time received event was blocked before it could be queued",
-				Buckets:   []float64{.002, .005, .01, .015, .025, .05, .1, .25, .5, .75, 1},
+				Namespace:                      Namespace,
+				Subsystem:                      SubsystemKVStore,
+				Name:                           "events_queue_seconds",
+				Help:                           "Duration in seconds of time received event was blocked before it could be queued",
+				Buckets:                        []float64{.002, .005, .01, .015, .025, .05, .1, .25, .5, .75, 1},
```

Comment:
At the moment some of the histograms specify the buckets explicitly, and others do not. This will make the native histograms inconsistent I think. From the [client_golang docs](https://github.com/prometheus/client_golang/blob/main/prometheus/histogram.go#L390-L395):
```
If Buckets is left as nil or set to a slice of length
zero, it is replaced by default buckets. The default buckets are
DefBuckets if no buckets for a native histogram (see below) are used,
otherwise the default is no buckets. (In other words, if you want to
use both reguler buckets and buckets for a native histogram, you have
to define the regular buckets here explicitly.)
```

So if Prometheus is configured to scrape native histograms, then for metrics with explicit buckets it will scrape both native and traditional histograms, but for others only the native histogram? Am I reading this right? In such case it would be nice to have it consistent IMO, i.e. define the buckets for all metrics explicitly.

Reply:
Oh, true. Very good catch. I will set `Buckets` as `DefBuckets` to the histograms that do not have bucket layout configured.

Reply:
I resolved that now.